Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+                    placement="bottom"
+                    link
+                >
+                    <div>
+                        <ReferenceField
+                            record={output_item}
+                            basePath="/sources"
+                            label="Sources"
+                            source="source_id"
+                            reference="sources"
+                            link="show"
+                        >
+                            <ChipConditionalLabel source="label" />
+                        </ReferenceField>
+                    </div>
+                </Tooltip>
+            ) : null}
+        </TableCell>
+    ));
+
+const InputSourceAssociation = ({ isRowExpanded, input_item }) => (
+    <TableCell
+        align="center"
+        rowSpan={isRowExpanded ? input_item.channels.length + 1 : 1}
+    >
+        {input_item.parent.type === 'source' ? (
+            <ReferenceField
+                record={input_item}
+                basePath="/sources"
+                label="Sources"
+                source="parent.id"
+                reference="sources"
+                link="show"
+            >
+                <ChipConditionalLabel source="label" />
+            </ReferenceField>
+        ) : input_item.parent.type === 'receiver' ? (
+            <ReferenceField
+                record={input_item}
+                basePath="/receivers"
+                label="Receivers"
+                source="parent.id"
+                reference="receivers"
+                link="show"
+            >
+                <ChipConditionalLabel source="label" />
+            </ReferenceField>
+        ) : null}
+    </TableCell>
+);
+
+const FillEmptyCellsForCollapseRow = ({ outputs, isColExpanded }) =>
+    outputs.map(([output_name, output_item]) => {
+        return isColExpanded(output_name) ? (
+            output_item.channels.map(() => <TableCell />)
+        ) : (
+            <TableCell />
+        );
+    });
+
+const InputChannelsMappingRows = ({
+    input_item,
+    input_nmos_name,
+    outputs,
+    isColExpanded,
+    mapping_disabeld,
+    handleMap,
+    isMapped,
+}) => {
+    return Object.entries(input_item.channels).map(([input_index, channel]) => (
+        <TableRow>
+            <TableCell size="small">{channel.label}</TableCell>
+            <>
+                {outputs.map(([output_nmos_name, output_item]) => {
+                    return isColExpanded(output_nmos_name) ? (
+                        output_item.channels.map((output_channel, index) => {
+                            return (
+                                <TableCell align="center">
+                                    <Tooltip
+                                        title={
+                                            <>
+                                                <Typography variant="body1">
+                                                    {'Input Channel: '}
```

Comment:
In the call today, we discussed that this tooltip should show the Output name as well as the channel, because the column heading might have been scrolled out of view.

Afterwards I realised this also applies to the row heading for the Input, especially for inputs with large numbers of channels. Therefore I think we need Input name as well.

Maybe something simple like 'Input: {input-name} {input-channel-label}' on one line and same for Output on the next line?